sort-test/src/main/java/com/hazelcast/jet/tests/sort/SortTest.java

```diff
@@ -0,0 +1,122 @@
+import static com.hazelcast.query.Predicates.alwaysTrue;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.IntStream.range;
+
+public class SortTest extends AbstractSoakTest {
+
+    private static final String SOURCE_MAP_NAME = "sortSourceMap";
+    private static final String SINK_LIST_NAME = "sortSinkList";
+
+    private static final int DEFAULT_ITEM_COUNT = 10_000;
+    private static final int LOG_JOB_COUNT_THRESHOLD = 100;
+    private static final int DEFAULT_SLEEP_AFTER_TEST_SECONDS = 5;
+
+    private IMap<Integer, String> sourceMap;
+    private IList<String> sinkList;
+    private List<String> expectedList;
+    private int itemCount;
+    private int sleepAfterTestSeconds;
+
+    public static void main(String[] args) throws Exception {
+        new SortTest().run(args);
+    }
+
+    @Override
+    protected void init(JetInstance client) throws Exception {
+        sourceMap = client.getMap(SOURCE_MAP_NAME);
+        sinkList = client.getList(SINK_LIST_NAME);
+
+        sourceMap.clear();
+        sinkList.clear();
+
+        itemCount = propertyInt("itemCount", DEFAULT_ITEM_COUNT);
+        sleepAfterTestSeconds = propertyInt("sleepAfterTestSeconds", DEFAULT_SLEEP_AFTER_TEST_SECONDS);
+
+        range(0, itemCount)
+                .mapToObj(i
+                        -> sourceMap.putAsync(i, UuidUtil.newUnsecureUuidString().substring(0, 4))
+                                .toCompletableFuture())
+                .collect(toList())
+                .forEach(f -> uncheckCall(f::get));
+        expectedList = sourceMap.values().stream().sorted().collect(toList());
+    }
+
+    @Override
+    protected void test(JetInstance client, String name) {
+        long begin = System.currentTimeMillis();
+        long jobCount = 0;
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            JobConfig jobConfig = new JobConfig();
+            jobConfig.setName("SortTest" + jobCount);
+            client.newJob(pipeline(), jobConfig).join();
+
+            assertEquals(itemCount, sinkList.size());
+            verifyList();
+            sinkList.clear();
+            if (jobCount % LOG_JOB_COUNT_THRESHOLD == 0) {
+                logger.info("Job count: " + jobCount);
+            }
+            jobCount++;
+            sleepSeconds(sleepAfterTestSeconds);
+        }
+        logger.info("Final job count: " + jobCount);
+    }
+
+    @Override
+    protected void teardown(Throwable t) {
+    }
+
+    private Pipeline pipeline() {
+        Pipeline p = Pipeline.create();
+        p.readFrom(Sources.map(sourceMap, alwaysTrue(), Map.Entry::getValue))
+                .sort(String::compareTo)
+                .writeTo(Sinks.list(sinkList));
+        return p;
+    }
+
+    private void verifyList() {
+        try {
+            for (int i = 0; i < itemCount; i++) {
+                assertEquals(expectedList.get(i), sinkList.get(i));
```

Comment:
instead of getting the item from sinkList one by one we can use `iterator` which under the hood gets all the items and wraps an iterator around them.